aptos-move/block-executor/src/executor.rs

```diff
@@ -143,55 +146,48 @@ where
+                output
+                    .aggregator_v1_write_set()
+                    .into_iter()
+                    .map(|(state_key, write_op)| (state_key, (write_op, None))),
+            ) {
+                if prev_modified_keys.remove(&k).is_none() {
+                    updates_outside = true;
                 }
+                versioned_cache
+                    .data()
+                    .write(k, idx_to_execute, incarnation, v);
+            }
 
-                for (k, v) in output.module_write_set().into_iter() {
-                    if prev_modified_keys.remove(&k).is_none() {
-                        updates_outside = true;
-                    }
-                    versioned_cache.modules().write(k, idx_to_execute, v);
+            for (k, v) in output.module_write_set().into_iter() {
+                if prev_modified_keys.remove(&k).is_none() {
+                    updates_outside = true;
                 }
+                versioned_cache.modules().write(k, idx_to_execute, v);
+            }
 
-                // Then, apply deltas.
-                for (k, d) in output.aggregator_v1_delta_set().into_iter() {
-                    if prev_modified_keys.remove(&k).is_none() {
-                        updates_outside = true;
-                    }
-                    versioned_cache.data().add_delta(k, idx_to_execute, d);
+            // Then, apply deltas.
+            for (k, d) in output.aggregator_v1_delta_set().into_iter() {
+                if prev_modified_keys.remove(&k).is_none() {
+                    updates_outside = true;
                 }
+                versioned_cache.data().add_delta(k, idx_to_execute, d);
+            }
 
-                for (id, change) in output.aggregator_v2_change_set().into_iter() {
-                    if !prev_modified_aggregators.remove(&id) {
-                        updates_outside = true;
-                    }
-
-                    // TODO: figure out if change should update updates_outside
-                    // versioned_cache.aggregators().
-                    match change {
-                        AggregatorChange::Create(value) => versioned_cache
-                            .aggregators()
-                            .create_aggregator(id, idx_to_execute, value),
-                        AggregatorChange::Apply(apply) => versioned_cache
-                            .aggregators()
-                            .record_apply(id, idx_to_execute, apply),
-                    }
+            for (id, change) in output.aggregator_v2_change_set().into_iter() {
+                if !prev_modified_aggregators.remove(&id) {
+                    updates_outside = true;
                 }
-            };
+
+                // TODO: figure out if change should update updates_outside
+                versioned_cache
+                    .aggregators()
+                    .record_change(id, idx_to_execute, change);
+            }
```

Comment:
this is the only change in this function, the rest is formatting.

I've moved the code there, but the fix was that Apply sometimes also needs to create a new entry as well

Reply:
oh for snapshots!